homeassistant/components/vacuum/xiaomi.py

```diff
@@ -0,0 +1,373 @@
+                        if value == speed][0]
+            return speed
+
+    @property
+    def fanspeed_list(self):
+        """Get the list of available fan speed steps of the vacuum cleaner."""
+        return list(FANSPEEDS.keys())
+
+    @property
+    def device_state_attributes(self):
+        """Return the specific state attributes of this vacuum cleaner."""
+        if self.vacuum_state is not None:
+            attrs = {
+                'Do not disturb':
+                    STATE_ON if self.vacuum_state.dnd else STATE_OFF,
+                # Not working --> 'Cleaning mode':
+                #    STATE_ON if self.vacuum_state.in_cleaning else STATE_OFF,
+                'Cleaning time': str(self.vacuum_state.clean_time),
+                'Cleaned area': self.vacuum_state.clean_area}
+            if self.vacuum_state.got_error:
+                attrs['Error'] = self.vacuum_state.error
+            return attrs
+
+        return {}
+
+    @property
+    def vacuum(self):
+        """Property accessor for vacuum object."""
+        if not self._vacuum:
+            from mirobo import Vacuum
+            _LOGGER.info("Initializing with host %s (token %s...)",
+                         self._host, self._token[:5])
+            self._vacuum = Vacuum(self._host, self._token)
+
+        return self._vacuum
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self._is_on
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_XIAOMI
+
+    @asyncio.coroutine
+    def _try_command(self, mask_error, func, *args, **kwargs):
+        """Call a vacuum command handling error messages."""
+        from mirobo import VacuumException
+        try:
+            yield from self.hass.async_add_job(partial(func, *args, **kwargs))
+            return True
+        except VacuumException as ex:
+            _LOGGER.error(mask_error, ex)
+            return False
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the vacuum on."""
+        is_on = yield from self._try_command(
+            "Unable to start the vacuum: %s", self.vacuum.start)
+        self._is_on = is_on
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn the vacuum off and return to home."""
+        yield from self.async_stop()
+        return_home = yield from self.async_return_to_base()
+        if return_home:
+            self._is_on = False
+
+    @asyncio.coroutine
+    def async_stop(self, **kwargs):
+        """Stop the vacuum cleaner."""
+        yield from self._try_command(
+            "Unable to stop: %s", self.vacuum.stop)
+
+    @asyncio.coroutine
+    def async_set_fanspeed(self, fanspeed: str, **kwargs):
```

Comment:
```py
def async_set_fan_speed(self, fan_speed: str, **kwargs):
```
Consider dropping the annotation on the positional argument since the annotation for the function is not complete, ie no return value annotation, and for consistency.